### PR TITLE
Boilerplated all necessary public interfaces

### PR DESCRIPTION
**Pull request by Kallb123**
Created the structure of the library overall. Mainly based upon the existing JS library, but with some C# changes.
I believe I have captured the most important parts, and can start to actually implement the methods.

Some of the methods are already implemented, because it's hard not to.

# Testing

1. Build the library. This provides a .dll file buried in the bin folder
2. Reference this library from another .NET application... The sample app for example!
3. Check to make sure the methods are exposed!

### CODE REVIEW

**Comment by amrc-benmorrow** [comment]
If this is for use in Unity I would strongly recommend using UniTask rather than native Task. Task is high-overhead in Unity due to running tasks on the thread pool. Although UniTask is primarily a Unity interface [it also exists as a NuGet package with the Unity-specific stuff removed](https://github.com/Cysharp/UniTask#net-core).

**Comment by Kallb123** [comment]
Good shout @amrc-benmorrow , didn't consider that. Can we use the UniTask nuget package and still have the library be crossplatform or would it require separate dotnet vs Unity builds?

**Comment by amrc-benmorrow** [comment]
I believe the nuget will work if you pull the DLL into Unity (I think that's the point) but honestly I've never tried it.

**Comment by Kallb123** [comment]
@amrc-benmorrow I also meant just using it in some other C# app like a .Net desktop app or ASPNET server? We could rename the library to unity-serviceclient if it's only unity-compatible though. Although our (very simple) sample test app wasn't going to be Unity.

File: utilities-dotnet/utilities-dotnet/Auth.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    public PostPrincipalBody(Guid uuid, string kerberos)
+    {
+        this.Uuid = uuid;
+        this.Kerberos = kerberos;
+    }
+}
+
+public struct PostAceBody
+{
+    public Guid Permission;
```
Comment (RBlackett) [inline utilities-dotnet/utilities-dotnet/Auth.cs:29]: This class is maybe missing comments on all it's public elements

**Review by RBlackett — APPROVED** [review]
Cracking start!

File: utilities-dotnet/utilities-dotnet/Auth.cs (hunk 1 above)
Comment (Kallb123) [reply utilities-dotnet/utilities-dotnet/Auth.cs:29]: Ah yes, I was looking to add them in as I fleshed out the Auth and got more familiar with it

**Pull request merged**